app/src/main/java/com/thanhclub/musicdemo/manager/SongPlayerManager.java

```diff
@@ -0,0 +1,67 @@
+package com.thanhclub.musicdemo.manager;
+
+
+import android.media.AudioManager;
+import android.media.MediaPlayer;
+
+import java.io.IOException;
+
+public class SongPlayerManager {
+    private MediaPlayer mediaPlayer;
+    public static final int PLAYER_IDEL = -1;
+    public static final int PLAYER_PREPARED = 0;
+    public static final int PLAYER_PLAYING = 1;
+    public static final int PLAYER_PAUSE = 2;
+    public static final int PLAYER_STOP = 3;
+    private int state;
+
+    public SongPlayerManager() {
+        state = PLAYER_IDEL;
+    }
+
+    public void setup(String path) {
+        try {
+            state = PLAYER_IDEL;
+            mediaPlayer = new MediaPlayer();
+            mediaPlayer.setDataSource(path);
+            mediaPlayer.setAudioStreamType(AudioManager.STREAM_MUSIC);
+            mediaPlayer.setOnPreparedListener(prepared);
+            mediaPlayer.prepare();
```

Comment:
e nên sử dụng prepareAsync cho stream
 For streams, you should call prepareAsync(), which returns immediately, rather than blocking until enough data has been buffered.
https://developer.android.com/reference/android/media/MediaPlayer.html#prepareAsync()